script.js

```diff
@@ -1,4 +1,398 @@
+var numPlayersPlacedBet = 0;
+// Function to get players' bets. Bets are stored in an array.
+var getBets = function (input) {
+  var index = 0;
+  var output = "";
+  if (numPlayersPlacedBet < playerName.length - 1) {
+    playerBets.push(input);
+    output = `${
+      playerName[numPlayersPlacedBet]
+    } is waging ${input} points. <br><br> ${
+      playerName[numPlayersPlacedBet + 1]
+    } please input your bet.`;
+    numPlayersPlacedBet += 1;
+    return output;
+  } else {
+    playerBets.push(input);
+    output = `${playerName[numPlayersPlacedBet]} is waging ${input} points. <br><br> All players have placed their bets. <br> Let the games begin!`;
+    gameMode = "shuffle";
+    return output;
+  }
+};
+
+// Function to update the players' current scores given the outcome of the game. First parameter is the index of the player, second parameter is boolean, if true, player wins the bet.
+var updateScores = function (playerIndex, playerWon) {
+  if (playerWon) {
+    playerScore[playerIndex] += Number(playerBets[playerIndex]);
+  } else {
+    playerScore[playerIndex] -= Number(playerBets[playerIndex]);
+  }
+};
+
+// Function to print the current points on all players.
+var showLeaderboard = function () {
+  var output = "";
+  for (var i = 0; i < playerName.length; i += 1) {
+    output += `${playerName[i]}: ${playerScore[i]} <br>`;
+  }
+  return `<br><br> ======LEADERBOARD====== <br> ${output} <br><br> To continue to the next round, ${playerName[0]} please input your bet amount for the next round.`;
+};
+
+// Function to deal the initial 2 cards to all players. Function takes in the number of players as an integer.
+var dealCards = function (numPlayers) {
+  var index = 0;
+  var inputArray = [];
+  while (index < numPlayers) {
+    inputArray.push([shuffledDeck[0], shuffledDeck[1]]);
+    shuffledDeck.splice(0, 2);
+    index += 1;
+  }
+  return inputArray;
+};
+
+// Function to total points of the player's hand. Takes in array of cards on player's hand.
+var calculateHand = function (playerHand) {
+  var outerIndex = 0;
+  var innerIndex = 0;
+  var totalScore = 0;
+  var numAce = 0;
+  var allPlayerPoints = [];
+  while (outerIndex < playerHand.length) {
+    while (innerIndex < playerHand[outerIndex].length) {
+      totalScore += playerHand[outerIndex][innerIndex].rank;
+      if (playerHand[outerIndex][innerIndex].name == "Ace") {
+        numAce += 1;
+      }
+      innerIndex += 1;
+    }
+    if (totalScore > 21 && numAce > 0) {
+      totalScore = 0;
+      for (var i = 0; i < playerHand[outerIndex].length; i += 1) {
+        if (playerHand[outerIndex][i].name == "Ace") {
+          playerHand[outerIndex][i].rank = 1;
+        }
+        totalScore += playerHand[outerIndex][i].rank;
+      }
+    }
+    allPlayerPoints.push(totalScore);
+    totalScore = 0;
+    innerIndex = 0;
+    outerIndex += 1;
+  }
+
+  return allPlayerPoints;
+};
```

Comment:
Nice work decomposing this logic into it's ow function! 

What would really help in your function comments is to add in what the function returns. 
Same goes for the other functions

E.g.
// Function to total points of the player's hand. 
// Takes in array of cards on player's hand. 
// Returns an array / string of player's hands